beacon_node/lighthouse_network/src/service/mod.rs

```diff
@@ -1431,6 +1451,11 @@ impl<AppReqId: ReqId, TSpec: EthSpec> Network<AppReqId, TSpec> {
                 };
                 self.build_response(id, peer_id, response)
             }
+            HandlerEvent::Close(_) => {
+                let _ = self.swarm.disconnect_peer_id(peer_id);
```

Comment:
Sorry for the noise @mxinden 

I think the new API does not fully allow the previous behavior. 

When emitting an event from the handler, this is received on `swarm.poll_next_unpin` without the information of the `connection_id`, so that this line could be `close_connection` instead. 

For LH this is good enough since we allow only one connection per peer, but it seems like a slight regression in connection management in libp2p. 

Now, if there is a way to achieve this "the correct way" (closing the specific connection) and this is an oversight on my part, happy to hear and use the alternative in this pr

Reply:
I am missing context here. What is `HandlerEvent::Close`? I can only find this type alias:

``` rust
/// Events the handler emits to the behaviour.
pub type HandlerEvent<Id, T> = Result<RPCReceived<Id, T>, HandlerErr<Id>>;
```

I am assuming that this is `std`'s `Result` which does not have a `Close` variant.

Reply:
I think what you are missing is a pull max. This type is no longer a result